storm-core/src/jvm/org/apache/storm/container/cgroup/CgroupCommon.java

```diff
@@ -0,0 +1,226 @@
+
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, NOTIFY_ON_RELEASE), flag ? "1" : "0");
+    }
+
+    @Override
+    public boolean getNotifyOnRelease() throws IOException {
+        return CgroupUtils.readFileByLine(Constants.getDir(this.dir, NOTIFY_ON_RELEASE)).get(0).equals("1") ? true : false;
+    }
+
+    @Override
+    public void setReleaseAgent(String command) throws IOException {
+        if (!this.isRoot) {
+            return;
+        }
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, RELEASE_AGENT), command);
+    }
+
+    @Override
+    public String getReleaseAgent() throws IOException {
+        if (!this.isRoot) {
+            return null;
+        }
+        return CgroupUtils.readFileByLine(Constants.getDir(this.dir, RELEASE_AGENT)).get(0);
+    }
+
+    @Override
+    public void setCgroupCloneChildren(boolean flag) throws IOException {
+        if (!this.cores.keySet().contains(SubSystemType.cpuset)) {
+            return;
+        }
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, CGROUP_CLONE_CHILDREN), flag ? "1" : "0");
+    }
+
+    @Override
+    public boolean getCgroupCloneChildren() throws IOException {
+        return CgroupUtils.readFileByLine(Constants.getDir(this.dir, CGROUP_CLONE_CHILDREN)).get(0).equals("1") ? true : false;
+    }
+
+    @Override
+    public void setEventControl(String eventFd, String controlFd, String... args) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        sb.append(eventFd);
+        sb.append(' ');
+        sb.append(controlFd);
+        for (String arg : args) {
+            sb.append(' ');
+            sb.append(arg);
+        }
+        CgroupUtils.writeFileByLine(Constants.getDir(this.dir, CGROUP_EVENT_CONTROL), sb.toString());
+    }
+
+    public Hierarchy getHierarchy() {
+        return hierarchy;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public String getDir() {
+        return dir;
+    }
+
+    public CgroupCommon getParent() {
+        return parent;
+    }
+
+    public Set<CgroupCommon> getChildren() {
+        return children;
+    }
+
+    public boolean isRoot() {
+        return isRoot;
+    }
+
+    public Map<SubSystemType, CgroupCore> getCores() {
+        return cores;
+    }
+
+    public void delete() throws IOException {
+        this.free();
+        if (!this.isRoot) {
+            this.parent.getChildren().remove(this);
+        }
+    }
+
+    private void free() throws IOException {
+        for (CgroupCommon child : this.children) {
+            child.free();
+        }
+        if (this.isRoot) {
+            return;
+        }
+        Set<Integer> tasks = this.getTasks();
+        if (tasks != null) {
+            for (Integer task : tasks) {
+                this.parent.addTask(task);
+            }
+        }
+        CgroupUtils.deleteDir(this.dir);
+    }
+
+    private void init() {
```

Comment:
Should we have `init()` up by the constructors where it's called? Seems silly to have it at the bottom.